service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java

```diff
@@ -631,4 +650,24 @@ private ResourceEntity persistResourceToZoneAndAssert(final ZoneEntity zoneEntit
         assertThat(this.resourceRepository.findOne(resourceEntity.getId()), equalTo(resource));
         return resourceEntity;
     }
+
+    @SuppressWarnings("unchecked")
+    static void deleteTwoLevelEntityAndParents(ZonableEntity e, final ZoneEntity zone,
+            final GraphGenericRepository repository) {
+        Set<Parent> parents = e.getParents();
+        for (Parent parent : parents) {
+            repository.delete(repository.getEntity(zone, parent.getIdentifier()));
+        }
+        repository.delete(e);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void deleteThreeLevelEntityAndParents(ZonableEntity e, ZoneEntity zone,
```

Comment:
All parameters should be `final`. Also introduce generics into the method signature to allow you to remove the `unchecked` warning:

```
private static <E extends ZonableEntity> void deleteThreeLevelEntityAndParents(final E zonableEntity, final ZoneEntity zone, final GraphGenericRepository<E> repository) {
    ...
}
```